fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -153,10 +156,6 @@ def __init__(
         self._trainable_param_to_rank: Dict[torch.Tensor, int] = {}
         self._reference_trainable_mask = list(map(_trainable, self._all_params))
 
-        # - keep track of the grads which have already been reduced
-        self._reduced_grads = 0
```

Comment:
no need for these anymore, it was there prior to using the Variable queue method

Reply:
we discovered that variable queue method and activation_checkpointing have some bad interactions. Basically, if you queue the finally callback in a checkpointed bwd, the callback can come before the overall bwd finishes. You may want to verify that's not affecting SDP.

Reply:
ohh, thanks for the headsup. It's already the case before this PR anyway, so I'm guessing that this can go through but to keep in mind, not sure how to grab that case ? (practically you can disable the buckets in that case for SDP, so there's at least one solution)